setuptools/config/_apply_pyprojecttoml.py

```diff
@@ -171,6 +173,10 @@ def _long_description(dist: "Distribution", val: _DictOrStr, root_dir: StrPath):
         _set_config(dist, "long_description_content_type", ctype)
 
     if file:
+        # Due to bad type narrowing, mypy sees file as `str | tuple[Any, ...]`
+        # and flipping the condition mypy sees val as `str | None`.
+        # Pyright gets both w/o issue. Can't use `tuple[()]()` until Python 3.9. So casting instead.
+        # https://github.com/python/mypy/issues/17273
```

Comment:
This still offers a path to remove the cast once Python 3.8 support is dropped, or if mypy improves its type narrowing for empty tuples (https://github.com/python/mypy/issues/17273)